snapcraft/plugins/kbuild.py

```diff
@@ -110,8 +110,10 @@ def do_base_config(self, config_path):
         # if kconfigfile is provided use that
         # otherwise use defconfig to seed the base config
         if self.options.kconfigfile is None:
-            # first run defconfig for setting baseline
-            self.run(self.make_cmd + self.options.kdefconfig)
+            # we need to run this with -j1, unit tests are a good defense here.
+            make_cmd = self.make_cmd.copy()
+            make_cmd[1] = '-j1'
+            self.run(make_cmd + self.options.kdefconfig)
```

Comment:
This seems a bit fragile. Why not refactor `self.make_cmd` into a function that accepts the number of build jobs as a parameter?